uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/Service/LeaveRequestTest.php

```diff
@@ -235,6 +235,24 @@ public function testDeleteSoftDeletesTheLeaveRequest() {
     $this->assertEquals(1, $leaveRequestRecord->is_deleted);
   }
 
+  public function testDeleteSoftDeletesAPublicHolidayLeaveRequest() {
+    AbsencePeriodFabricator::fabricate([
+      'start_date' => CRM_Utils_Date::processDate('2017-01-01'),
+      'end_date' => CRM_Utils_Date::processDate('2017-12-31')
+    ]);
+
+    $publicHoliday = new PublicHoliday();
+    $publicHoliday->date = CRM_Utils_Date::processDate('2017-10-10');
+
+    $publicHolidayLeaveRequest = PublicHolidayLeaveRequestFabricator::fabricate($this->leaveContact, $publicHoliday);
+    $this->getLeaveRequestServiceWhenCurrentUserIsAdmin()->delete($publicHolidayLeaveRequest->id);
+
+    $publicHolidayLeaveRequestRecord = new LeaveRequest();
+    $publicHolidayLeaveRequestRecord->id = $publicHolidayLeaveRequest->id;
+    $publicHolidayLeaveRequestRecord->find(true);
```

Comment:
These three lines of code can be replaced with `LeaveRequest::findById($publicHolidayLeaveRequest->id);`

Reply:
LeaveRequest::findById() is a modified version that only returns results for leave request with is_deleted = 0.